dali/kernels/signal/dct/dct_gpu.cu

```diff
@@ -29,11 +29,49 @@ namespace kernels {
 namespace signal {
 namespace dct {
 
+struct Lifter {
+  DALI_HOST_DEV
+  explicit Lifter(float coeff): coeff_(coeff) {}
+
+  DALI_HOST_DEV
+  float operator()(float val) { return val * coeff_; }
+
+  const float coeff_;
+};
+
+struct IdLifter {
+  DALI_HOST_DEV
+  float operator()(float val) { return val; }
+};
+
+template <bool nonzero>
+struct LiftersTable {};
+
+template <>
+struct LiftersTable<true> {
+  explicit LiftersTable(const float *lifter_coeffs): coeffs_(lifter_coeffs) {}
+
+  DALI_HOST_DEV
+  Lifter lifter(int idx) const {
+    return Lifter(coeffs_[idx]);
+  }
+
+ private:
+  const float *coeffs_ = nullptr;
+};
+
+template <>
+struct LiftersTable<false> {
+  DALI_HOST_DEV
+  IdLifter lifter(int) {return IdLifter{}; }
+};
+
+
 // The kernel processes data with the shape reduced to 3D.
 // Transform is applied over the middle axis.
-template <typename OutputType, typename InputType>
+template <typename OutputType, typename InputType, bool nonzero>
```

Comment:
I'd do either
1)
```suggestion
template <typename OutputType, typename InputType, typename LifterTable>
```
2) or remove the zero version and rely on the bool directly:
```
sample.output[output_idx] = HasLifter ? coeff * out_val : out_val;
```
where HasLifter is your `nonzero` template argument

Reply:
I've remove LifterTabble
used HasLifter param